### PR TITLE
auto-animate doesn't consider line shapes #661 

### LINKED ISSUE #661: auto-animate doesn't consider line shapes
Lines and arrows are not auto-animated

### PR DESCRIPTION
**Pull request by initialshl**
Fixes #661 

This PR implements auto animation for straight line shapes. 
Elbow connector lines and curves are not supported as they are much more difficult to implement.

Every line shape begins at top left corner and ends at bottom right corner of its boundary box. Its rotation is always zero, and uses flip instead. Due to this behavior, some trigonometry is used to calculate the final dimensions and rotation with respect to the initial properties. 

Exisiting FTs: AutoAnimateSuccessfully, AnimateInSlideTest, Flaky_AutoAnimateWithCopyPasteSuccessfully passed.
Added FTs: AutoAnimateStraightLinesSuccessfully, AnimateInSlideStraightLinesTest passed.

ezgif com-gif-maker

### CODE REVIEW

**Comment by kai33** [comment]
Hi Heng Le, thanks for this cool fix!
Could you help to add a piece of functional test to guard this change?

File: PowerPointLabs/PowerPointLabs/DefaultMotionAnimation.cs [hunk 1]
```diff
@@ -9,12 +9,15 @@ namespace PowerPointLabs
     class DefaultMotionAnimation
     {
 #pragma warning disable 0618
+
+        private const float ArrowheadLength = 6.0f;
```
Comment (kai33) [inline PowerPointLabs/PowerPointLabs/DefaultMotionAnimation.cs]: why a magic number 6 here?

Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/DefaultMotionAnimation.cs]: Oh thanks for pointing out the bad practice. I realized that arrowhead can have different sizes, which causes the rotation pivot to be off from the center. The new commit fixes the pivot issue by grouping a transparent line in the opposite direction.

File: PowerPointLabs/PowerPointLabs/FrameMotionAnimation.cs [hunk 2]
```diff
@@ -157,5 +177,80 @@ private static void AddFrameAnimationEffects(PowerPointSlide animationSlide, Pow
...
+            List<PowerPoint.Shape> toGroup = new List<PowerPoint.Shape> { line, transparentLine };
+            PowerPoint.Shape newLine = animationSlide.ToShapeRange(toGroup).Group();
+
+            animationSlide.TransferAnimation(tempAnimationHolder, newLine);
+            tempAnimationHolder.Delete();
+
+            return newLine;
+        }
+
+        private static double GetLineAngle(PowerPoint.Shape shape, bool acute = false)
```
Comment (kai33) [inline PowerPointLabs/PowerPointLabs/FrameMotionAnimation.cs]: the acute is for?

Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/FrameMotionAnimation.cs]: Its supposedly a flag on whether the method should return the angle as an acute angle, turns out i can reduce the number of trigo functions and its not needed anymore (removed in latest commit).

**Comment by initialshl** [comment]
> Hi Heng Le, thanks for this cool fix!
> Could you help to add a piece of functional test to guard this change?

@kai33 Hi, I have added the functional tests for AutoAnimate and AnimateInSlide for straight lines, thanks!

**Pull request merged**